### ISSUE TITLE
Empty google_bigquery_table schema crashes Terraform

### ISSUE THREAD

**Issue by smaslennikov** [hashicorp/terraform-provider-google#7302]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
Terraform v0.13.2
+ provider registry.terraform.io/hashicorp/google v3.39.0
```

### Affected Resource(s)

* `google_bigquery_table`

### Terraform Configuration Files

https://github.com/sadasystems/terraform-google-bq-table-crash

### Debug Output

https://github.com/sadasystems/terraform-google-bq-table-crash/blob/master/sample_run

### Expected Behavior

The second apply run with no errors or suggested changes.

### Actual Behavior

Terraform crashes.

### Steps to Reproduce

1. `terraform apply`
1. `terraform apply` again

### Important Factoids

I realize this may not be an expected scenario, but it's still probably not ideal that Terraform plain crashes in its wake.

Looks like actually [leaving the schema blank](https://github.com/sadasystems/terraform-google-bq-table-crash/commit/d1af2588373a7c72e08bd5e8e380787e1843c0a4#diff-7a370d8342e7203b805911c92454f0f4L14) instead of setting it to an empty list works out fine.

### References

* #0000

**Comment by c2thorn** [comment]
Thanks for providing a configuration and debug log @smaslennikov! It made it very easy to reproduce. I made a fix for this, hopefully it will go out in v3.42

**Comment by smaslennikov** [comment]
wohoo, thanks @c2thorn!
